vermouth/pdb/nwalign.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+
+Sequence alignment with the Needleman-Wunsch algorithm
+
+Designed to check the SEQRES entry of an input PDB versus the residues present
+
+Adapted from
+https://github.com/zaneveld/full_spectrum_bioinformatics/blob/master/content/08_phylogenetic_trees/needleman_wunsch_alignment.ipynb
```

Comment:
This is a problem. That code is CC BY-NC-SA 4.0 licensed, which is /way/ more restrictive than the Apache license we carry. See https://creativecommons.org/licenses/by-nc-sa/4.0/

Reply:
I don't see the issue here; what Chris referenced is an educational resource and if I'm not mistaken but that's for Chris to confirm he did not copy the code in verbatim and / or large chunks of it. The algorithm itself is not restricted by the Licence

Reply:
Any derivative of that code has to be either CC BY-NC-SA or GPL licensed. The algorithm itself is indeed not covered, but derivative implementations are. The fact that it's an educational resource is irrelevant.

Reply:
I guess the question is more at what point is it a derivative code?

Reply:
could we ask for special permission?

Reply:
> I guess the question is more at what point is it a derivative code?

IANAL, so no idea. I err on the side of caution here

> could we ask for special permission?

We could actually. However, don't you think it'll be less work to build off of an MIT/BSD/Apache implementation? I also sent an old ab-initio implementation of my own to @csbrasnett earlier today.

Reply:
I'm looking at the code @pckroon sent earlier now, it won't be too much effort to adapt so I'll sort it out in the next commit

Reply:
I'm fine with either way; I didn't look closely enough at the code to know how difficult that is. In general any work is always more work than just sending a mail/request. But in the end it's the choice of @csbrasnett

Reply:
It's not their algorithm. Just simple, somewhat convoluted Needleman-Wunsch. I have an (own) implementation of it that could be put in instead. Can dig that up. No licensing issues then.